akka-remote/src/main/scala/akka/remote/artery/Association.scala

```diff
@@ -396,7 +411,34 @@ private[remote] class Association(
     }
   }
 
-  override def toString(): String =
+  // FIXME: Compression table must be owned by each channel instead of having a global one
+  private def createOutboundCompressionTable(remoteAddress: Address): OutboundCompression = {
+    if (transport.provider.remoteSettings.ArteryCompressionSettings.enabled) {
+      val compression = new OutboundCompressionImpl(transport.system, remoteAddress)
+      log.info("Creating Outbound compression table ({}) to [{}]", compression.hashCode, remoteAddress)
+      compression
+    } else NoOutboundCompression
+  }
+
+  /*
+   * This proxy uses the current associationStates compression table, which is reset for a new incarnation.
+   * This way the same outgoing stream will switch to using the new table without the need of restarting it.
+   */
+  object CurrentAssociationStateOutboundCompressionProxy extends OutboundCompression {
```

Comment:
This allows the active streams to access the "right" compression table, i.e. if we have a new associationState